packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -207,7 +207,16 @@ export function compareCellPositionsUsingTombstones(
 		// If both changesets know of both cells, but we've been asked to compare different cells,
 		// Then either the changesets they originate from do not represent the same context,
 		// or the ordering of their cells in inconsistent.
-		assert(false, 0x8a0 /* Inconsistent cell ordering */);
+		// The only exception to this is when we're composing anonymous changesets in a transaction.
+		assert(
+			oldMarkCell.revision === undefined && newMarkCell.revision === undefined,
+			0x8a0 /* Inconsistent cell ordering */,
+		);
+		// We are composing anonymous changesets in a transaction. The new changeset is creating a cell in a gap
+		// where the old changeset knows of some now empty cell. We order the new cell relative to the old cell in a
+		// way that is consistent with its tie-breaking behavior should the old cell be concurrently re-filled.
+		// Since only tie-break left is supported at the moment, the new cell comes first.
+		return CellOrder.NewThenOld;
```

Comment:
Shouldn't both changesets know about the empty old cell? I think that would mean that we should use NewThenOld even if the new cell has tie-break right.

Reply:
No, this is the scenario prompting this codepath:
`compose(compose(insert@0, delete@0), insert@0)`